#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_v8_db_update.dpatch by Tim Retout <tim@retout.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix updating to v8 of the f-spot photo database.
## DP: This was applied upstream as SVN revision 3748.

@DPATCH@
commit c4aae33d95c62a274bc1e494ca7894883614ac4d
Author: sdelcroix <sdelcroix@772769d8-d925-0410-89eb-a6ffa0d40526>
Date:   Wed Mar 12 08:54:02 2008 +0000

    2008-03-12  Stephane Delcroix  <sdelcroix@novell.com>
    
    	* Updater.cs: fixes the dp updating v8 where version names are int.
    	bgo #521656. Backporting the change from r3747.
    
    
    git-svn-id: http://svn.gnome.org/svn/f-spot/branches/FSPOT_0_4_2_STABLE@3748 772769d8-d925-0410-89eb-a6ffa0d40526

diff --git a/ChangeLog b/ChangeLog
index 3bc48f0..de13645 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2008-03-12  Stephane Delcroix  <sdelcroix@novell.com>
+
+	* Updater.cs: fixes the dp updating v8 where version names are int.
+	bgo #521656. Backporting the change from r3747.
+
 2008-02-12  Lorenzo Milesi <maxxer@yetopen.it>
 
 	* src/FormClient.cs:
diff --git a/src/Updater.cs b/src/Updater.cs
index 0595322..c21ced1 100644
--- a/src/Updater.cs
+++ b/src/Updater.cs
@@ -162,13 +162,14 @@ namespace FSpot.Database {
 					string uri = photo_uri.Scheme + "://" + 
 						photo_uri.Host + 
 						System.IO.Path.GetDirectoryName (photo_uri.AbsolutePath) + "/" +
-						name_without_extension + " (" + (reader [2] as string) + ")" + extension;
+						name_without_extension + " (" + (reader [2]).ToString () + ")" + extension;
+
 					ExecuteNonQuery (new DbCommand (
 						"INSERT INTO photo_versions (photo_id, version_id, name, uri) " +
 						"VALUES (:photo_id, :version_id, :name, :uri)",
 						"photo_id", Convert.ToUInt32 (reader [0]),
 						"version_id", Convert.ToUInt32 (reader [1]),
-						"name", (string)(reader [2]),
+						"name", (reader [2]).ToString (),
 						"uri", uri));
 				}
 
